### PR TITLE
Issue 76 - enable phpcs and set up code coverage upload

### PR DESCRIPTION
**Pull request by gesinn-it-ilm**
This PR is made in reference to: #76

This PR contains:

- phpcs added to composer.json
- reported sniffs excluded and cleared
- upload code coverage enabled
- added matrix for MW 1.40
- tests updated, deprecated staff removed
- badges added to README.md file

### CODE REVIEW

**Comment by gesinn-it-gea** [comment]
@YOUR1 would you please quickly review this PR?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -24,6 +24,13 @@ jobs:
             database_image: "mariadb:10"
             coverage: false
             experimental: true
+          - mediawiki_version: '1.40'
+            smw_version: dev-master
+            php_version: 8.1
+            database_type: mysql
+            database_image: "mariadb:11.2"
+            coverage: true
+            experimental: true
```
Comment (gesinn-it-gea) [inline .github/workflows/ci.yml:33]: for all cases where tests are passing, experimental can be set to false

Comment (gesinn-it-ilm) [reply .github/workflows/ci.yml]: @gesinn-it-gea done.

**Comment by YOUR1** [comment]
> @YOUR1 would you please quickly review this PR?

Seems good to me!

**Pull request merged**